Source/WebCore/svg/SVGPathByteStream.h

```diff
@@ -39,71 +40,158 @@ namespace WebCore {
 class SVGPathByteStream : public CanMakeSingleThreadWeakPtr<SVGPathByteStream> {
     WTF_MAKE_FAST_ALLOCATED;
 public:
-    typedef Vector<uint8_t> Data;
-    typedef Data::const_iterator DataIterator;
-
-    SVGPathByteStream() { }
+    class Data final : public RefCounted<Data> {
+    public:
+        friend class NeverDestroyed<Data, MainThreadAccessTraits>;
+
+        using Bytes = Vector<uint8_t>;
+
+        static Ref<Data> create(Bytes&& bytes)
+        {
+            return adoptRef(*new Data(WTFMove(bytes)));
+        }
+
+        void updatePath(const Path& path) const
+        {
+            m_path = path;
+        }
+
+        static Ref<Data> empty()
+        {
+            static MainThreadNeverDestroyed<Data> singleton;
+            return Ref { singleton.get() };
+        }
+
+        const Bytes& bytes() const { return m_bytes; }
+
+        void append(uint8_t byte)
+        {
+            m_path = { };
+            m_bytes.append(byte);
+        }
+
+        void append(std::span<const uint8_t> bytes)
+        {
+            m_path = { };
+            m_bytes.append(bytes);
+        }
+
+        void append(const Bytes& other)
+        {
+            m_path = { };
+            m_bytes.appendVector(other);
+        }
+
+        const Path& path() const { return m_path; }
+
+        Ref<Data> copy() const
+        {
+            return adoptRef(*new Data(*this));
+        }
+
+        friend bool operator==(const Data& a, const Data& b)
+        {
+            return a.m_bytes == b.m_bytes;
+        }
+
+        unsigned size() const { return m_bytes.size(); }
+        bool isEmpty() const { return m_bytes.isEmpty(); }
+
+    private:
+        Data(Bytes&& bytes)
+            : m_bytes(WTFMove(bytes))
+        {
+        }
+
+        Data() = default;
+        Data(const Data& data)
```

Comment:
`= default` would do the same ?

Reply:
The reason why it does not work is it is `RefCounted` class, which we do not copy m_refCount.